pkg/services/datasources/service/datasource_service.go

```diff
@@ -559,3 +565,18 @@ func awsServiceNamespace(dsType string) string {
 		panic(fmt.Sprintf("Unsupported datasource %q", dsType))
 	}
 }
+
+func (s *Service) fillWithSecureJSONData(ctx context.Context, cmd *models.UpdateDataSourceCommand, ds *models.DataSource) error {
+	decrypted, err := s.DecryptedValues(ctx, ds)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range decrypted {
+		if _, ok := cmd.SecureJsonData[k]; !ok {
```

Comment:
Can `cmd.SecureJsonData` be nil/not initialized?

Reply:
Sanity check added on ffad198